messages/handler/handler.go

```diff
@@ -37,88 +87,146 @@ type Getter interface {
 	Get(messages.Get) (interface{}, *messages.Failure, error)
 }
 
+type Deleter interface {
+	Delete(messages.Delete) (*messages.Deleted, *messages.Failure, error)
+}
+
 // Aggregate is the interface to handle Aggregate messages
 type Aggregate interface {
 	Creater
 	Modifier
+	Deleter
 	Getter
 }
 
-// onAggregate is a struct containing aggregates for registered types.
+// onMessage is a struct containing aggregates for registered types.
 //
 // The messages.MessageType is normalized like what.on e.g. create.target
-// onAggregate tries to parse the given messages to messages.Create,
+// onMessage tries to parse the given messages to messages.Create,
 // messages.Modify, messages.Get then tries to find via MessageType the
 // Aggregate via handler.
-type onAggregate struct {
-	handler map[string]Aggregate
+type onMessage struct {
+	lookup map[string]Holder
 }
 
-func (mh onAggregate) On(message []byte) (interface{}, error) {
+func asResponse(t string, d interface{}) *connection.SendResponse {
+	return &connection.SendResponse{
+		Topic: t,
+		MSG:   d,
+	}
+}
+
+func (mh onMessage) On(topic string, message []byte) (*connection.SendResponse, error) {
```

Comment:
I personally think this method is to long, and should be refactored into multiple smaller ones.

Reply:
Refactored with: a384f09e69e4a083f72fef3eeee4390ab521ed5c